### PR TITLE
add checkOverflow, clean up style and test code

### PR DESCRIPTION
**Pull request by wlammen**
Mostly clean up and minor improvements

### CODE REVIEW

**Comment by benjub** [comment]
> 1. Improve on some names to make their meaning clearer;

I'm all for it.  Maybe `fatl`-->`fatal` within variable names and file names ?

**Comment by wlammen** [comment]
Hi @benjub , did not expect to hear at this place.  Are you going join the programmers front?

**Comment by wlammen** [comment]
Perhaps I should split this PR into several commits, each following one idea, so they can better be checked.  Maybe tomorrow.

File: src/mmfatl.c [hunk 1]
```diff
@@ -10,7 +10,14 @@
  * conditions (corrupt state, out of memory).
  *
  * C99 (https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf)
- * compatible.
+ * compatible.  For a check enter the directory where build.sh is located,
+ * create a subdirectory named 'build', concatenate the following command lines
+ * into a single line, and run:
+ *
+ * cd build && gcc -I. -I../src -I.. -std=c99 -pedantic
+ * -DINLINE=inline -DTEST_ENABLE -c -o mmfatl.o ../src/mmfatl.c
```
Comment (digama0) [inline src/mmfatl.c:18]:
```suggestion
 * create a subdirectory named 'build', and run:
 *
 * cd build && gcc -I. -I../src -I.. -std=c99 -pedantic \
 *   -DINLINE=inline -DTEST_ENABLE -c -o mmfatl.o ../src/mmfatl.c
```

**Comment by digama0** [comment]
Let's postpone the FATL thing for a pass which renames all the file names to not be 8.3-compatible.

**Pull request merged**